compiler/rustc_mir_transform/src/lower_intrinsics.rs

```diff
@@ -191,6 +191,31 @@ impl<'tcx> MirPass<'tcx> for LowerIntrinsics {
                             terminator.kind = TerminatorKind::Goto { target };
                         }
                     }
+                    sym::transmute => {
+                        let dst_ty = destination.ty(local_decls, tcx).ty;
+                        let Ok([arg]) = <[_; 1]>::try_from(std::mem::take(args)) else {
+                            span_bug!(
+                                terminator.source_info.span,
+                                "Wrong number of arguments for transmute intrinsic",
+                            );
+                        };
+
+                        // Always keep the cast, because that lets CTFE and codegen
+                        // generate better error messages when it's done wrong.
```

Comment:
This comment makes sense as part of this PR, but someone who sees it in the future will probably not get why it says "keep".

```suggestion
                        // Always emit the cast, even if we transmute to an uninhabited type,
                        // because that lets CTFE and codegen
                        // generate better error messages when such a transmute actually ends
                        // up reachable.
```

Reply:
Oh, I meant "keep" as in "preserve the transmute from the Call to an Assign", but I like your wording better.  I'll update it.